### PR TITLE
FEATURE: Add setting & preference for search sort default order

### PR DESCRIPTION
**Pull request by martin-brennan**
This commit adds a new `search_default_sort_order` site setting,
set to "relevance" by default, that controls the default sort order
for the full page /search route.

If the user changes the order in the dropdown on that page, we remember
their preference automatically, and it takes precedence over the site
setting as a default from then on. This way people who prefer e.g.
Latest Post as their default can make it so.

### CODE REVIEW

File: app/assets/javascripts/admin/addon/services/search-preferences-manager.js
Comment (davidtaylorhq) [inline app/assets/javascripts/admin/addon/services/search-preferences-manager.js:1]: Should probably be `javascripts/discourse/app/services`, not `javascripts/admin/addon/services` ?

File: app/assets/javascripts/discourse/app/templates/full-page-search.hbs [hunk 1]
```diff
@@ -145,7 +145,7 @@
             <ComboBox
               @value={{this.sortOrder}}
               @content={{this.sortOrders}}
-              @onChange={{action (mut this.sortOrder)}}
+              @onChange={{action this.setSortOrder}}
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/templates/full-page-search.hbs:148]:
```suggestion
              @onChange={{this.setSortOrder}}
```

File: spec/system/search_spec.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
-describe "Search", type: :system do
+describe "Search", type: :system, js: true do
```
Comment (CvX) [inline spec/system/search_spec.rb:3]: is `js: true` needed?

**Review by CvX — APPROVED** [review]

File: app/assets/javascripts/discourse/app/templates/full-page-search.hbs (hunk 1 above)
Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/templates/full-page-search.hbs:148]: I will never get used to these action changes :sweat_smile:

File: spec/system/search_spec.rb (hunk 2 above)
Comment (martin-brennan) [reply spec/system/search_spec.rb:3]: Ah yeah probs not, will remove

**Pull request merged**